flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlReplaceTableAs.java

```diff
@@ -0,0 +1,278 @@
+
+        if (isIfNotExists()) {
+            throw new SqlValidateException(
+                    getParserPosition(),
+                    errorMsg + " syntax does not support IF NOT EXISTS statements yet.");
+        }
+
+        if (isTemporary()) {
+            throw new SqlValidateException(
+                    getParserPosition(),
+                    errorMsg + " syntax does not support to create temporary table yet.");
+        }
+
+        if (getColumnList().size() > 0) {
+            throw new SqlValidateException(
+                    getParserPosition(),
+                    errorMsg + " syntax does not support to specify explicit columns yet.");
+        }
+
+        if (getWatermark().isPresent()) {
+            throw new SqlValidateException(
+                    getParserPosition(),
+                    errorMsg + " syntax does not support to specify explicit watermark yet.");
+        }
+        if (getPartitionKeyList().size() > 0) {
+            throw new SqlValidateException(
+                    getParserPosition(),
+                    errorMsg + " syntax does not support to create partitioned table yet.");
+        }
+        if (getFullConstraints().stream().anyMatch(SqlTableConstraint::isPrimaryKey)) {
+            throw new SqlValidateException(
+                    getParserPosition(),
+                    errorMsg + " syntax does not support primary key constraints yet.");
+        }
+    }
+
+    public SqlNode getAsQuery() {
+        return asQuery;
+    }
+
+    public boolean isCreateOrReplace() {
+        return isCreateOrReplace;
+    }
+
+    public SqlIdentifier getTableName() {
+        return tableName;
+    }
+
+    public SqlNodeList getColumnList() {
+        return columnList;
+    }
+
+    public SqlNodeList getPropertyList() {
+        return propertyList;
+    }
+
+    public SqlNodeList getPartitionKeyList() {
+        return partitionKeyList;
+    }
+
+    public List<SqlTableConstraint> getTableConstraints() {
+        return tableConstraints;
+    }
+
+    public Optional<SqlWatermark> getWatermark() {
+        return Optional.ofNullable(watermark);
+    }
+
+    public Optional<SqlCharStringLiteral> getComment() {
+        return Optional.ofNullable(comment);
+    }
+
+    public boolean isIfNotExists() {
+        return ifNotExists;
+    }
+
+    public boolean isTemporary() {
+        return isTemporary;
+    }
+
+    /** Returns the column constraints plus the table constraints. */
+    public List<SqlTableConstraint> getFullConstraints() {
+        return SqlConstraintValidator.getFullConstraints(tableConstraints, columnList);
+    }
+
+    @Override
+    public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
+        if (isCreateOrReplace) {
+            writer.keyword("CREATE");
+            writer.keyword("OR");
+        }
+        writer.keyword("REPLACE");
+        if (isTemporary()) {
```

Comment:
since `isTemporary` is not supported, we can remove it since we also remove the partition & columns in here.